### ISSUE TITLE
Custom gradients at layer level instead of at function level

### ISSUE THREAD

**Issue by AlphardHilbert** [tensorflow/tensorflow#25453]
I am implementing a customized layer in tensorflow. However, I failed to figure out a way to customize gradients in layers. I would like to add a function to calculate the gradients to weights and gradients to input by myself. Is there any way to do so?

The code should look like this
```
class MyDenseLayer(tf.keras.layers.Layer):
    def __init__(self, num_outputs):
        super(MyDenseLayer, self).__init__()
        self.num_outputs = num_outputs

    def build(self, input_shape):
        self.kernel = self.add_variable("kernel", shape=[int(input_shape[-1]), self.num_outputs])

    def call(self, input):
        return tf.matmul(input, self.kernel)
    
    # The function to calculate the gradients by myself
    def _backprop(self,y_delta,x):
        w_delta=some_function(y_delta,x)
        x_delta=some_function2(y_delta,x)
        return w_delta, x_delta
````

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by facaiy** [comment]
Automatically closing due to lack of recent activity. Feel free to reopen it if necessary.
